core/src/main/antlr4/io/delta/sql/parser/DeltaSqlBase.g4

```diff
@@ -89,6 +91,11 @@ statement
     | .*?                                                               #passThrough
     ;
 
+temporalClause
+    : FOR? (SYSTEM_VERSION | VERSION) AS OF version=(INTEGER_VALUE | STRING)
+    | FOR? (SYSTEM_TIME | TIMESTAMP) AS OF timestamp=STRING
```

Comment:
Wait... you are the timestamp as a string, not an expression. that means generate expression with timestamp arithmetic wont work ...right?

Reply:
Currently we don't have support for parsing an arbitrary expression. One idea is to take the expression as string here and parse it and evaluate it using the Catalyst expression utilities as part of the `RestoreCommandStatement` creation from parser output. 

I am thinking we can make this improvement once this PR is committed.